File: Excel/800-899/835/835 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/800-899/835/835 Max Salary.xlsx"
input = read_excel(path, range = "A2:C12")
test  = read_excel(path, range = "E2:G6")

result = input %>%
  fill(Dept) %>%
  slice_max(order_by = Salary, n = 1, with_ties = T, by = Dept) %>%
  mutate(Dept = if_else(row_number() == 1, Dept, NA_character_), .by = Dept) %>%
  rename(`Max Salary` = Salary)

all.equal(result, test)
# [1] TRUE

File: Excel/800-899/835/835 Challenge.py
import pandas as pd

path = "800-899/835/835 Max Salary.xlsx"

input = pd.read_excel(path, usecols="A:C", skiprows=1, nrows=11)
test = pd.read_excel(path, usecols="E:G", skiprows=1, nrows=4).rename(columns=lambda x: x.replace('.1', ''))

input['Dept'] = input['Dept'].ffill()
result = input[input.groupby('Dept')['Salary'].transform('max') == input['Salary']].reset_index(drop=True)
result['Dept'] = result['Dept'].where(~result['Dept'].duplicated())
result = result.rename(columns={'Salary': 'Max Salary'})

print(result.equals(test)) # True